BuildLoader.py

```diff
@@ -1119,16 +1120,17 @@ def pre_build(self):
         # generate size variables
         for each in fsp_list:
             fsp_bin  = os.path.join(self._fv_dir, "%s.bin" % each)
-            if self._board.FSP_IMAGE_ID:
+            if self._board.HAVE_FSP_BIN:
+                if self._board.FSP_IMAGE_ID:
                     imageid = get_fsp_image_id (fsp_bin)
                     if self._board.FSP_IMAGE_ID != imageid:
-                            raise Exception ('Expecting FSP ImageId: %s, but got %s !' % (self._board.FSP_IMAGE_ID, imageid))
-            revision = get_fsp_revision (fsp_bin)
-            if revision < self._board.MIN_FSP_REVISION:
-                    raise Exception ('Required minimum FSP revision is 0x%08X, but current revision is 0x%08X !' %
-                                (self._board.MIN_FSP_REVISION, revision))
+                        raise Exception ('Expecting FSP ImageId: %s, but got %s !' % (self._board.FSP_IMAGE_ID, imageid))
+                revision = get_fsp_revision (fsp_bin)
+                if revision < self._board.MIN_FSP_REVISION:
+                        raise Exception ('Required minimum FSP revision is 0x%08X, but current revision is 0x%08X !' %
+                                    (self._board.MIN_FSP_REVISION, revision))
             setattr(self._board, '%s_SIZE' % each, get_fsp_size(fsp_bin) if self._board.HAVE_FSP_BIN else 0)
-            setattr(self._board, '%s_UPD_SIZE' % each, get_fsp_upd_size(fsp_bin) if self._board.HAVE_FSP_BIN else 0)
+            setattr(self._board, '%s_UPD_SIZE' % each, get_fsp_upd_size(fsp_bin) if self._board.HAVE_FSP_BIN else 1)

```

Comment:
Do we need 'else 1'?

Reply:
Yes,  we need it.  If the UPD size is 0, the build will fail because:
UINT8    FspmUpd[FixedPcdGet32 (PcdFSPMUpdSize)];  
We have to keep it a non-zero value.

Reply:
Got it. Thanks for the explanation.